codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -39,22 +39,30 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
 
 /**
  * Simple implementation of {@link Http2Connection}.
  */
 public class DefaultHttp2Connection implements Http2Connection {
 
+    private static final Comparator<Http2Stream> STREAMID_COMPARATOR = new Comparator<Http2Stream>() {
+        @Override
+        public int compare(Http2Stream o1, Http2Stream o2) {
+            return o1.id() - o2.id();
+        }
+    };
+
     private final Set<Listener> listeners = new HashSet<Listener>(4);
     private final IntObjectMap<Http2Stream> streamMap = new IntObjectHashMap<Http2Stream>();
     private final ConnectionStream connectionStream = new ConnectionStream();
-    private final Set<Http2Stream> activeStreams = new LinkedHashSet<Http2Stream>();
+    private final Set<Http2Stream> activeStreams = new ConcurrentSkipListSet<Http2Stream>(STREAMID_COMPARATOR);
```

Comment:
@buchgr - That makes sense...hum I will try to think of something that is not hacky...What about instead of exposing an iterator use something like a `ActiveStreamProcessor` (similar to `ByteBufProcessor`) so the connection has visibility/control over modifications while an iteration is happening.  When an iteration is in process the connection could enable all modification events to be queued, and then process those events once all iterations have ended?